snapd-glib/requests/snapd-json.c

```diff
@@ -300,6 +302,8 @@ _snapd_json_get_date_time (JsonObject *object, const gchar *name)
     if (timezone == NULL)
         timezone = g_time_zone_new_local ();
 
+    if (string_format != NULL)
+        *string_format = g_strdup(value);
```

Comment:
Why is this being returned as a string - isn't the remaining text the number of nanoseconds that we can return in a guint64 or similar?

Reply:
The idea is to put the exact same string that the server sent to us, with the whole date and time, when asking for the next notice. Also, the nanoseconds are passed in the string as a decimal value after the seconds, and the trailing zeros are removed, so it would require some pre- and post-processing, both for adding the trailing zeros when storing it as nanoseconds in the guint64, and also to take into account zeros immediately after the dot when converting it again to the string sent to the server. I already tried using a gdouble, but sometimes there is rounding errors in the conversion, resulting in the server sending the last notice over and over again. Also, this is only for "internal consumption": it is only to be used when asking for new notices, to specify the date of the last notice received, so it makes sense to just "paste it as-is" in the request message.

Reply:
Not processing the value has a bad code smell, while I understand in this case it's being used more as a token than a timestamp (also an indication of some bad architecture upstream) we should still be able to parse it. I don't think the overhead of the pre- and post-processing is significant in this case to warrant a workaround.

You should definitely not use a gdouble as it's a decimal fraction, but store it in an unsigned integer.

Reply:
Done.